src/main/java/version02/Product.java

```diff
@@ -0,0 +1,9 @@
+package version02;
+
+public interface Product {
+   int getId();
+   void setId(int id);
```

Comment:
Warum gibt es `setId` und `setName` am interface? Die werden nirgends aufgerufen und in der einen Klasse in `setId` nicht mal implementiert. Ich würde die setter kompett löschen. `Product`s wären dann immutable. Evtl. könnte man aus den Klassen dann auch records machen.